core/nginx/letsencrypt.py

```diff
@@ -45,33 +43,20 @@
 # Wait for nginx to start
 time.sleep(5)
 
-class MyRequestHandler(SimpleHTTPRequestHandler):
-    def do_GET(self):
-        if self.path == '/.well-known/acme-challenge/testing':
-            self.send_response(204)
-        else:
-            self.send_response(404)
-        self.send_header('Content-Type', 'text/plain')
-        self.end_headers()
-
-def serve_one_request():
-    with HTTPServer(("127.0.0.1", 8008), MyRequestHandler) as server:
-        server.handle_request()
-
 # Run certbot every day
 while True:
     while True:
         hostname = os.environ['HOSTNAMES'].split(',')[0]
         target = f'http://{hostname}/.well-known/acme-challenge/testing'
-        thread = Thread(target=serve_one_request)
-        thread.start()
-        r = requests.get(target)
-        if r.status_code != 204:
-            log.critical(f"Can't reach {target}!, please ensure it's fixed or change the TLS_FLAVOR.")
-            time.sleep(5)
-        else:
-            break
-        thread.join()
+        try:
+            r = requests.get(target)
+            if r.status_code != 204:
+                log.critical(f"Can't reach {target}!, please ensure it's fixed or change the TLS_FLAVOR.")
+                time.sleep(5)
+            else:
+                break
+        except Exception as e:
+            log.error(f"Exception while fetching {target}!", exc_info = e)
```

Comment:
we need some delay here:
```suggestion
            log.error(f"Exception while fetching {target}!", exc_info = e)
            time.sleep(15)
```

Reply:
That's c11815b3